### ISSUE TITLE
[Feature Request] Detect unqualified column references in window function PARTITION BY/ORDER BY when JOINs present

### ISSUE THREAD

**Issue by SonChauNgoc** [sqlfluff/sqlfluff#7395]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

When a query contains JOINs and window functions, using SELECT aliases in `PARTITION BY` or `ORDER BY` clauses can cause **silent, incorrect results** if the alias name matches a column name in a joined table.

This happens because of SQL's logical execution order: window functions are evaluated (step 5) *before* SELECT aliases are created (step 6). When the database encounters an unqualified reference like `PARTITION BY id`, it resolves to an actual column from a joined table (`t2.id`) rather than the intended alias.

**The bug is insidious because:**
- ✅ Passes all current sqlfluff rules (including RF02)
- ✅ Compiles and runs without errors
- ❌ Returns **wrong data** silently

**Example of the problem:**

```sql
SELECT DISTINCT
    t1.col1 AS id,              -- Creates alias 'id'
    CASE 
        WHEN t1.col3 IS NULL 
        THEN LAG(t1.col3) IGNORE NULLS OVER (
            PARTITION BY id ORDER BY dt  -- ❌ 'id' resolves to t2.id, not the alias!
        )
        ELSE t1.col3 
    END AS col3_filled
FROM table1 AS t1
LEFT JOIN table2 AS t2 ON t1.col1 = t2.id  -- t2 has column named 'id'
```

When `t2` has no matching rows, `t2.id = NULL` for all rows, causing **all rows to end up in the same NULL partition** — and `LAG` pulls data from completely unrelated rows.

### Use case

**What I want to happen:**

A new rule (or enhancement to RF02) that warns when:
1. A query contains `JOIN` clauses
2. A window function uses **unqualified column references** in `PARTITION BY` or `ORDER BY` clauses

**Buggy code (should be flagged):**

```sql
SELECT DISTINCT
    t1.col1 AS id,
    LAG(t1.col3) IGNORE NULLS OVER (
        PARTITION BY id ORDER BY dt  -- ❌ Unqualified references
    ) AS col3_filled
FROM table1 AS t1
LEFT JOIN table2 AS t2 ON t1.col1 = t2.id;
```

**Fixed code (should pass):**

```sql
SELECT DISTINCT
    t1.col1 AS id,
    LAG(t1.col3) IGNORE NULLS OVER (
        PARTITION BY t1.col1 ORDER BY t1.col2  -- ✅ Table-qualified references
    ) AS col3_filled
FROM table1 AS t1
LEFT JOIN table2 AS t2 ON t1.col1 = t2.id;
```

**Possible implementation approaches:**
- **Option A:** Enhance `RF02` (`references.qualification`) to inspect window function clauses
- **Option B:** Create a new rule like `RF03` / `references.window_qualification` or `ambiguity.window_partition`

**We created a custom rule as a workaround** — happy to share the implementation if helpful for the core project.

### Dialect

- **Primary:** Snowflake
- **Likely affected:** Any dialect that allows alias references in window functions (Snowflake extends standard SQL to allow this)

This bug was discovered in a production Snowflake environment where queries passed all linting but returned incorrect results on specific days when joined tables had no matching data.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
